README.md

```diff
@@ -75,8 +75,8 @@ The error object can be passed along with the original source to `Parsimmon.form
     any character except one of the characters `"a"`, `"b"`, or `"c"`,
     and will yield the same.
   - `Parsimmon.regex(/myregex/, group=0)` is a parser that expects the stream
-    to match the given regex, and yields the given match group, or the
-    entire match.
+    to match the given regexp, and yields the given match group, or the
+    entire match. The regexp may only use the following flags: `imu`. `Parsimmon.regex` will throw an error for any other flag. The regexp is also wrapped in `^(?: regexp )` because it doesn't make sense to match something further away than the next character. Greedy regexp should be wrapped like this: `( regexp )+` (e.g. `Parsimmon.regex(/(a+|a*b)+/)`).
```

Comment:
@michaelficarra @wavebeem 

Ah, yeah sorry about this. This whole thing has gone in circles because of me because I haven't been paying attention to what I'm doing or how I'm saying it.

It started off by me filing an #68 about how global flags couldn't be used to match tokens greedily. I didn't elaborate well enough what I meant by this, so I'll try to clarify. I was working on a parser (I think it was for CSS) and I was reading it's grammar and translating to parsimmon. One of it's tokens was this:

`[+\-]?([0-9]+|[0-9]*\.[0-9]+)(e[+-]?[0-9]+)?` - matches a number such as `+1.4`

When I've written similar regex for tokenization in other parsers (like O'Caml, Jison, Eulalie, etc), I haven't ran into issues with it not fully matching the token (this is what I meant by "greedy"). But in parsimmon, it won't fully match the above:

```
console.log(p.regex(/[+\-]?([0-9]+|[0-9]*\.[0-9]+)(e[+-]?[0-9]+)?/).parse('+1.4'))
{ status: false,  index: { offset: 2, line: 1, column: 3 },  expected: [ 'EOF' ] }
```

I was trying to churn out this CSS parser lib quickly so I could get to working on another project, so in my frustration I just tried the global flag to get the full token to match, saw it worked, gave up on the lib, and then filed an issue, completely forgetting that it's stateful and that I've been burned bad in the past by it. I didn't  even notice what Michael meant by disallowing `g` until wavebeam mentioned that it's stateful and then I remembered.

And actually wrapping the regex in `()+` only worked in the initial example because it's a hack since parsimmon matches the first capturing group by default. However, upon closer inspection I notice that it would match two tokens such as `-1.4+2.5` and treat it as a single token, so this hack would be useless anyway. So @wavebeem , you were right initially - I don't think there's a way to get it to fully match without `g`. But I haven't ran into this kind of issue with other lexers until now. 

@michaelficarra , I know you've worked extensively with parsers before. I don't want to bother you with this, but if you know off the top of your head how I can get that regex to match as a single token without splitting it up into multiple regex's and using `p.seq` I'd definitely be grateful. It might be obvious and I'm just still brainfarting on this.